src/main/java/org/kuali/kfs/kim/document/IdentityManagementRoleDocument.java

```diff
@@ -261,6 +261,11 @@ public void addMember(final KimDocumentRoleMember member) {
         final String roleMemberId = nextSeq.toString();
         member.setRoleMemberId(roleMemberId);
         setupMemberRspActions(member);
+
+        if (ObjectUtils.isNull(member.getActiveFromDate())) {
+            member.setActiveFromDate(getDateTimeService().getCurrentTimestamp());
```

Comment:
If the functionals want Role documents to auto-initialize the Active From Date when necessary, I would suggest moving such initialization to the `doRouteStatusChange()` method override so that it's performed when the document finishes routing, not when the member is first added to the document.